src/main/java/sqlline/Application.java

```diff
@@ -182,7 +180,11 @@ public String getVersion() throws Exception {
     if (inputStream != null) {
       // If not running from a .jar, pom.properties will not exist, and
       // inputStream is null.
-      properties.load(inputStream);
+      try {
+        properties.load(inputStream);
+      } catch (IOException e) {
+        e.printStackTrace();
```

Comment:
Better error handling?

Reply:
The issue here is that this method `sqlline.Application#getVersion` is used as  default value for `VERSION` property. 
That means in case of usage anything like `throw new AnyRuntimeException` the application will stop if the error is present, while in case of just printing message it will continue working

Reply:
Can we just log the exception?

Reply:
do you mean just message like
```        
System.err.println(e.getCause() == null
     ? e.getMessage() : e.getCause().getMessage());
```
or what?

Reply:
Nope, like adding logger and than use logger.warn

Reply:
May be I do not understand you correctly but currently there is no loggers in sqlline. Even debug/error methods in `sqlline.SqlLine` do output to `System.out/err`. May be it make sense to file a separate issue for logging

Reply:
I roll backed the behavior as it was before with throwing exception and handling at `SqlLine#getInfoMessage` level via `handleException`. The two things I updated is usage of try for resources and `UNKNOWN_VERSION` constant.
For `SqlLinePropertiesEnum` there was introduced a private class to init version value with `printstacktrace` handling as no other could be applied there (`throw AnyException` is not allowed while enum values initializations)

Reply:
Decided to stay at current version as mentioned here 
https://github.com/julianhyde/sqlline/pull/174#discussion_r227368161